tests/simple_run.rs

```diff
@@ -34,56 +38,71 @@ fn should_run_off_chain_worker() {
 		let header = chain_client.header(None).compat().await.unwrap();
 		println!("{:?}", header);
 
-	});
+		test.produce_blocks(15);
 
-	test.produce_blocks(15);
-	test.assert_log_line("db", "best = true");
+		// test.assert_log_line("db", "best = true");
+	});
 }
 
 #[test]
 fn should_read_state() {
 	// given
-	let mut test = test::deterministic(test::node::<Runtime>().start());
+	let node = start_node::<runtime::opaque::Block, runtime::RuntimeApi, Executor, _>(&[], spec_factory).unwrap();
+	let mut test = test::deterministic::<Runtime>(node);
 	type Balances = pallet_balances::Module<Runtime>;
 
 	test.produce_blocks(1);
 
-	let alice = Sr25519Keyring::Alice.pair();
+	let alice = Sr25519Keyring::Charlie.pair();
 	let bob = Sr25519Keyring::Bob.pair();
-	let signer = PairSigner::new(alice.clone());
-
-	let rpc_handlers = test.rpc_handler();
-
-	let alice_balance = test.with_state(|| Balances::free_balance(MultiSigner::from(alice.public()).into_account()));
-
-	test.tokio_runtime().block_on_std(async {
-		let client = ClientBuilder::<DefaultNodeRuntime>::new()
-			.set_client(subxt::SubxtClient::new(rpc_handlers))
-			.build()
-			.await
-			.unwrap();
-
-		client
-			.transfer(
-				&signer,
-				&Address::from(MultiSigner::from(bob.public()).into_account()),
-				8900000000000000,
-			)
-			.await
-			.expect("failed to transfer funds");
+
+	let signer = Signer::<Runtime, RuntimeKeyType>::all_accounts()
+		// only use alice' account for signing
+		.with_filter(vec![alice.public().into()]);
+
+	let (bob_balance, alice_balance) = test.with_state(|| {
+		let events = frame_system::Module::<Runtime>::events();
+		log::info!("{:#?}", events);
+		(
+			Balances::free_balance(MultiSigner::from(bob.public()).into_account()),
```

Comment:
Seems we always only use `bob.public()` or `alice.public()` can't we get it at the declaration level to simplify the code?